src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AmazonWebServicesCredentialsBinding.java

```diff
@@ -87,14 +88,23 @@ public String getSecretKeyVariable() {
         return secretKeyVariable;
     }
 
+    @DataBoundSetter
+    public void setStsTokenDuration(@Nullable Integer stsTokenDuration) {
+        this.stsTokenDuration = stsTokenDuration;
+    }
+
     @Override
     protected Class<AmazonWebServicesCredentials> type() {
         return AmazonWebServicesCredentials.class;
     }
 
     @Override
     public MultiEnvironment bind(@Nonnull Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-        AWSCredentials credentials = getCredentials(build).getCredentials();
+        AmazonWebServicesCredentials credentialRetriever = getCredentials(build);
+        if (stsTokenDuration != null) {
+            credentialRetriever.setStsTokenDuration(stsTokenDuration);
```

Comment:
This will potentially change the configuration in the credentials store, and thus change the behavior at least for all builds in the current session which do not specify this parameter.

Reply:
ah, I didn't realise this. Thanks for letting me know.